utils/src/test/kotlin/storage/LocalFileStorageTest.kt

```diff
@@ -0,0 +1,144 @@
+ * License-Filename: LICENSE
+ */
+
+package com.here.ort.utils.storage
+
+import com.here.ort.utils.safeMkdirs
+
+import io.kotlintest.shouldBe
+import io.kotlintest.shouldNotThrowAny
+import io.kotlintest.shouldThrow
+import io.kotlintest.specs.WordSpec
+
+import java.io.BufferedReader
+import java.io.File
+import java.io.FileNotFoundException
+
+class LocalFileStorageTest : WordSpec() {
+    private fun storage(block: (LocalFileStorage, File) -> Unit) {
+        val directory = createTempDir(javaClass.simpleName)
+        val storage = LocalFileStorage(directory)
+        block(storage, directory)
+    }
+
+    init {
+        "Creating the storage" should {
+            "succeed if the directory exists" {
+                shouldNotThrowAny {
+                    LocalFileStorage(createTempDir(javaClass.simpleName))
+                }
+            }
+
+            "succeed if the directory does not exist and must be created" {
+                val directory = createTempDir(javaClass.simpleName)
+                val storageDirectory = directory.resolve("create/storage")
+
+                LocalFileStorage(storageDirectory)
+
+                storageDirectory.isDirectory shouldBe true
+            }
+
+            "fail if the directory is a file" {
+                val storageDirectory = createTempFile()
+
+                shouldThrow<IllegalArgumentException> {
+                    LocalFileStorage(storageDirectory)
+                }
+            }
+        }
+
+        "Reading a file" should {
+            "succeed if the file exists" {
+                storage { storage, directory ->
+                    val file = directory.resolve("file-exists")
+                    file.writeText("file-exists")
+
+                    val content = storage.read("file-exists").bufferedReader().use(BufferedReader::readText)
+
+                    content shouldBe "file-exists"
+                }
+            }
+
+            "fail if the file does not exist" {
+                storage { storage, _ ->
+                    shouldThrow<FileNotFoundException> {
+                        storage.read("file-does-not-exist")
+                    }
+                }
+            }
+
+            "fail if the requested path is not inside the storage directory" {
+                storage { storage, _ ->
+                    shouldThrow<IllegalArgumentException> {
+                        storage.read("../file")
+                    }
+                }
+            }
+        }
+
+        "Writing a file" should {
+            "succeed if the file does not exist" {
+                storage { storage, directory ->
+                    storage.write("target/file", "content".toByteArray().inputStream())
+
+                    val file = directory.resolve("target/file")
+
+                    file.isFile shouldBe true
+                    file.readText() shouldBe "content"
+                }
+            }
+
+            "succeed if the file does exist" {
```

Comment:
This test is just a copy of the above one, which is why is was not discovered that overwriting actually does not work.

Reply:
Thanks, fixed the test. But overwriting actually did work, it just wasn't tested properly.